src/DataTable/sorting.ts

```diff
@@ -24,11 +24,112 @@ export function transition(direction: Exclude<SortDirection, 'NONE'>): Exclude<S
   return SortDirection.ASC
 }
 
+// ----------------------------------------------------------------------------
+// Sort strategies
+// ----------------------------------------------------------------------------
+
+/**
+ * A sort strategy for comparing any two values
+ */
 export function basic<T>(a: T, b: T) {
-  return a === b ? 0 : a < b ? 1 : -1
+  return a === b ? 0 : a < b ? -1 : 1
+}
+
+/**
+ * A sort strategy for comparing two `Date` values. Also includes support for
+ * values from `Date.now()`
+ */
+export function datetime(a: Date | number, b: Date | number): number {
+  const timeA = a instanceof Date ? a.getTime() : a
+  const timeB = b instanceof Date ? b.getTime() : b
+  return timeA > timeB ? 1 : timeA < timeB ? -1 : 0
+}
+
+/**
+ * Compare two numbers using alphanumeric, or natural order, sorting. This
+ * sorting function breaks up the inputs into groups of text and numbers and
+ * compares the different sub-groups of each to determine the order of a set of
+ * strings
+ *
+ * @see https://en.wikipedia.org/wiki/Natural_sort_order
+ */
+export function alphanumeric(inputA: string, inputB: string): number {
+  const groupsA = getAlphaNumericGroups(inputA)
+  const groupsB = getAlphaNumericGroups(inputB)
+
+  while (groupsA.length !== 0 && groupsB.length !== 0) {
+    const a = groupsA.shift()
+    const b = groupsB.shift()
+
+    // If the two groups are equal, move on to the next set of groups
+    if (a === b) {
+      continue
+    } else if (typeof a === 'string' && typeof b === 'string') {
+      // If both groups are strings, compare them using the current locale
+      return a.localeCompare(b)
+    } else if (typeof a === 'number' && typeof b === 'number') {
+      // If both groups are numbers, compare them numerically
+      return a > b ? 1 : -1
+    } else if (typeof a === 'number' && typeof b === 'string') {
+      // Sort numbers before strings
+      return -1
+    } else if (typeof a === 'string' && typeof b === 'number') {
+      // Sort numbers before strings
+      return 1
+    } else if (a === undefined || b === undefined) {
```

Comment:
What if `a` or `b` are `null`?

Reply:
Great call out, I swear any `undefined` call site has me anxious with `null` too 😂 Thankfully `Array.prototype.shift()` should be `string | number | undefined` so we don't have to worry about `null` (I hope 🤞)